packages/babel-plugin/src/utils/manipulate-template-literal.ts

```diff
@@ -0,0 +1,143 @@
+import traverse from '@babel/traverse';
+import * as t from '@babel/types';
+import { cssAfterInterpolation } from '@compiled/css';
+
+import type { Metadata } from '../types';
+
+import { getPathOfNode } from './ast';
+
+const conditionalPaths: ['consequent', 'alternate'] = ['consequent', 'alternate'];
+
+/**
+ * Returns `true` if the CSS property or pseudo classes/pseudo elements
+ * are defined within a template element (the string pieces of the template literal)
+ * rather than in the expression.
+ * Eg color: ${({ isPrimary }) => (isPrimary ? 'green' : 'red')};
+ * @param node
+ */
+export const isCssPropertyInTemplateElement = (node: t.TemplateElement): boolean => {
+  const value = node.value.raw;
+
+  return value.includes(':');
+};
+
+/**
+ * TODO: this is a temporary workaround so that we don't evaluate expressions that may throw an error.
+ * It should be removed after addressing https://github.com/atlassian-labs/compiled/issues/1081
+ *
+ * Returns `true` if
+ * - there is another conditional expression at the root of the current template literal
+ *   with a conditional expression to evaluate OR
+ * - there is a nested logical expression within the current template literal
+ *
+ * @param node
+ * @param meta {Metadata} The current metadata to use to find the parent node
+ */
+export const hasNestedTemplateLiteralsWithConditionalRules = (
```

Comment:
Let me know if there is a better way to traverse the nodes to check nested conditional rules.
I couldn't find anything to traverse bottom-up :(